accounts/models.py

```diff
@@ -0,0 +1,70 @@
+from django.contrib.auth.models import AbstractBaseUser, PermissionsMixin, BaseUserManager
+from django.db import models
+from django.utils import timezone
+from django.utils.translation import ugettext_lazy as _
+
+
+class RapidProUserManager(BaseUserManager):
+
+    def _create_user(self, email, password,
+                     is_staff, is_superuser, **extra_fields):
+        """
+        Creates and saves a User with the given email and password.
+        """
+        now = timezone.now()
+        if not email:
+            raise ValueError('The given email must be set')
+        email = self.normalize_email(email)
+        user = self.model(email=email,
+                          is_staff=is_staff, is_active=True,
+                          is_superuser=is_superuser, last_login=now,
+                          date_joined=now, **extra_fields)
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_user(self, email, password=None, **extra_fields):
+        return self._create_user(email, password, False, False,
+                                 **extra_fields)
+
+    def create_superuser(self, email, password, **extra_fields):
+        return self._create_user(email, password, True, True,
+                                 **extra_fields)
+
+
+class RapidProUser(AbstractBaseUser, PermissionsMixin):
+    """
+    A fully featured User model with admin-compliant permissions that uses
+    a full-length email field as the username.
+
+    Email and password are required. Other fields are optional.
+    """
+    email = models.EmailField(_('email address'), max_length=254, unique=True)
+    username = models.CharField(_('username'), max_length=254, blank=True)
```

Comment:
Why do we not just have a field named "username" with definition of email field? Why have both of these fields?